addons/modules/functions/fnc_moduleSpawnReinforcements.sqf

```diff
@@ -0,0 +1,136 @@
+#include "script_component.hpp"
+/*
+ * Author: mharis001
+ * Zeus module function to spawn reinforcements.
+ *
+ * Arguments:
+ * 0: Vehicle Type <STRING>
+ * 1: Infantry Types <ARRAY>
+ * 2: Spawn Position <ARRAY>
+ * 3: LZ Position <ARRAY>
+ * 4: RP Position <ARRAY>
+ * 5: RTB and Despawn <BOOL>
+ * 6: Insertion Method <NUMBER>
+ *   - 0: Land, 1: Paradrop, 2: Fastrope
+ * 7: Infantry Behaviour <NUMBER>
+ *   - 0: Default, 1: Relaxed, 2: Cautious, 3: Combat
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * ["B_MRAP_01_F", ["B_Soldier_F"], [0, 0, 0], [100, 100, 100], [200, 200, 200], true, 0, 0] call zen_modules_fnc_moduleSpawnReinforcements
+ *
+ * Public: No
+ */
+
+#define AIRCRAFT_HEIGHT 100
+#define WAYPOINT_RADIUS 30
+
+params ["_vehicleType", "_infantryTypes", "_spawnPosition", "_positionLZ", "_positionRP", "_despawnVehicle", "_insertionMethod", "_infantryBehaviour"];
+
+// Determine the direction the vehicle should face on spawn
+private _direction = _spawnPosition getDir _positionLZ;
+
+// Adjust spawn position for air vehicles
+private _isAir = _vehicleType isKindOf "Air";
+private _placement = "NONE";
+
+if (_isAir) then {
+    _spawnPosition set [2, AIRCRAFT_HEIGHT];
+    _placement = "FLY";
+};
+
+// Spawn vehicle at the start position and set its direction
+private _vehicle = createVehicle [_vehicleType, _spawnPosition, [], 0, _placement];
+_vehicle setDir _direction;
+
+private _vehicleConfig = configFile >> "CfgVehicles" >> _vehicleType;
+
+// Set a sufficient initial velocity for planes
+if (getText (_vehicleConfig >> "simulation") == "airplanex") then {
+    _vehicle setVelocity [sin _direction * 100, cos _direction * 100, 0];
+};
+
+// Determine the side of the units from the vehicle
+private _side = [getNumber (_vehicleConfig >> "side")] call BIS_fnc_sideType;
+
+// Create the vehicle crew
+private _vehicleGroup = createGroup [_side, true];
+createVehicleCrew _vehicle;
+
+crew _vehicle joinSilent _vehicleGroup;
+_vehicleGroup addVehicle _vehicle;
+
+_vehicleGroup selectLeader commander _vehicle;
+_vehicleGroup allowFleeing 0;
+
+// Force aircraft to fly at the correct height and ignore surroundings
+if (_isAir) then {
+    _vehicleGroup setBehaviour "CARELESS";
+    _vehicle flyInHeight AIRCRAFT_HEIGHT;
+};
+
+// Increase the skill of the driver/pilot for better driving/flying
+driver _vehicle setSkill 1;
+
+// Spawn the infantry units and move them into the vehicle's cargo positions
+private _infantryGroup = createGroup [_side, true];
+
+{
+    private _unit = _infantryGroup createUnit [_x, [0, 0, 0], [], 0, "CAN_COLLIDE"];
+    _unit moveInCargo _vehicle;
+} forEach _infantryTypes;
```

Comment:
If function get to many cargo for the vehicle... the units are at [0,0,0]
```
{
    private _unit = _infantryGroup createUnit [_x, [0, 0, 0], [], 0, "CAN_COLLIDE"];
    _unit assignAsCargoIndex [heli, _forEachIndex];
    _unit moveInCargo _vehicle;

    if (_vehicle getCargoIndex _unit != _forEachIndex) then {deleteVehicle _unit};
} forEach _infantryTypes;
```
or
```
{
    private _unit = _infantryGroup createUnit [_x, [0, 0, 0], [], 0, "CAN_COLLIDE"];
    _unit assignAsCargo _vehicle;
    if !(_unit moveInAny _vehicle) then {deleteVehicle _unit};
} forEach _infantryTypes;
```

Reply:
The module doesn't allow more units to be selected than spots are available in the vehicle, so this is a non-issue.